### DISCUSSION TITLE
Allow multiple instances of GDevelop to share user preferences

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3648] [Enhancement ideas (and their technical discussions)]
## Description

If a user accidentally opens two instances of GDevelop, the second instance cannot load user preferences, so it uses the default values.  When the user closes the second instance, it will save those default values and override the real ones.

[Chat about this topic](https://discordapp.com/channels/258623956158906368/258623956158906368/835209574961250324)

## Solution suggested

- When GDevelop starts, perform a check to see if another instance is open.  If so, close the new instance.

This might be a relevant solution:
https://github.com/electron/electron/blob/master/docs/api/app.md#apprequestsingleinstancelock

## Alternatives considered

- Provide the ability for multiple instances to be open (for copy/paste between projects, working on two games, etc).  This method would need to solve the problem of default settings overriding previously saved preferences.

**Comment by arthuro555** [comment]
I don't think disallowing multiple windows is a real solution, it is just avoiding the real problem. 
Instead, we should finally add support for multiple instances at once using `app.on('second-instance')` to make a new window with another instance of the react app. If we move reading the preferences to the main process, the main process will be shared and the renderers will not conflict while accessing the preferences at the same time.

**Comment by 4ian** [comment]
Indeed I think the real solution here is to add support for multiple instances with preferences properly shared between them :) 
Or a single instance with support for multiple windows.
